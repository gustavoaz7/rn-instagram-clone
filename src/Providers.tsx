import React, { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components/native';
import { Provider } from 'react-redux';
import { theme as stylesTheme } from './styles/theme';
import { store as reduxStore } from './redux/store';

type TProvidersProps = {
  children: ReactNode;
  store?: typeof reduxStore;
  theme?: typeof stylesTheme;
};

export function Providers({
  children,
  store = reduxStore,
  theme = stylesTheme,
}: TProvidersProps): JSX.Element {
  return (
    <Provider store={store}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </Provider>
  );
}
