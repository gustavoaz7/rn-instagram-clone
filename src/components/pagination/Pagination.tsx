import React from 'react';
import { ViewProps } from 'react-native';
import styled from 'styled-components/native';

export type TPaginationProps = {
  total: number;
  current: number;
  style?: ViewProps['style'];
};

export const Pagination = ({
  total,
  current,
  style,
}: TPaginationProps): JSX.Element => {
  return (
    <Container style={style}>
      {[...Array(total)].map((_, i) => (
        // eslint-disable-next-line react/no-array-index-key
        <Dot key={`dot-${i}`} active={i === current} testID="Pagination-Dot" />
      ))}
    </Container>
  );
};

const Container = styled.View`
  flex-direction: row;
`;

const Dot = styled.View<{ active: boolean }>`
  ${({ theme: { color }, active }) => `
    width: ${active ? 6 : 5}px;
    height: ${active ? 6 : 5}px;
    border-radius: ${active ? 6 : 5}px;
    background-color: ${active ? color.lightBlue : color.gray};
    margin: 0 2px;
  `}
`;
