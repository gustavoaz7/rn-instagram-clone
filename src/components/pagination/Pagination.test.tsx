import React from 'react';
import { render } from '@testing-library/react-native';
import { Pagination } from './Pagination';
import { Providers } from '../../Providers';

describe('components - Pagination', () => {
  const options = { wrapper: Providers };
  const props = { total: 4, current: 1 };

  it('renders', () => {
    render(<Pagination {...props} />, options);
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<Pagination {...props} />, options);

    expect(toJSON()).toMatchSnapshot();
  });
});
