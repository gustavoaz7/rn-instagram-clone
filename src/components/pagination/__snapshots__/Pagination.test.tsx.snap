// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - Pagination matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    active={false}
    style={
      Array [
        Object {
          "backgroundColor": "#8e8e8e",
          "borderRadius": 5,
          "height": 5,
          "marginBottom": 0,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 0,
          "width": 5,
        },
      ]
    }
  />
  <View
    active={true}
    style={
      Array [
        Object {
          "backgroundColor": "#0095f6",
          "borderRadius": 6,
          "height": 6,
          "marginBottom": 0,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 0,
          "width": 6,
        },
      ]
    }
  />
  <View
    active={false}
    style={
      Array [
        Object {
          "backgroundColor": "#8e8e8e",
          "borderRadius": 5,
          "height": 5,
          "marginBottom": 0,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 0,
          "width": 5,
        },
      ]
    }
  />
  <View
    active={false}
    style={
      Array [
        Object {
          "backgroundColor": "#8e8e8e",
          "borderRadius": 5,
          "height": 5,
          "marginBottom": 0,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 0,
          "width": 5,
        },
      ]
    }
  />
</View>
`;
